builtin/providers/aws/config_test.go

```diff
@@ -0,0 +1,275 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// Grab any existing AWS keys and preserve. In some tests we'll unset these, so
+// we need to have them and restore them after
+var k = os.Getenv("AWS_ACCESS_KEY_ID")
+var s = os.Getenv("AWS_SECRET_ACCESS_KEY")
+var to = os.Getenv("AWS_SESSION_TOKEN")
```

Comment:
Note these variables become global to the entire package during testing. Might be better to just have `unsetEnv` return a struct of the original values that `resetEnv` can consume to put them back.

Or you could avoid intermediate variables altogether by having unsetEnv return the reset func.

```go
resetEnv := unsetEnv(t)
defer resetEnv(t)
```

```
func unsetEnv(...) {
  oldKey := os.Getenv("KEY")
  return func(...) {
    os.Setenv("KEY", oldKey)
  }
}
```